ArduCopter/AP_Arming.cpp

```diff
@@ -352,6 +358,61 @@ bool AP_Arming_Copter::motor_checks(bool display_failure)
     return true;
 }
 
+// check Motor count
+bool AP_Arming_Copter::motor_count_check()
+{
+    // Number of motors in the frame
+    int8_t motorNum = 0;
+    switch ((AP_Motors::motor_frame_class)copter.g2.frame_class.get()) {
+    case AP_Motors::MOTOR_FRAME_HELI:
+        motorNum = 0;
+        break;
+    case AP_Motors::MOTOR_FRAME_QUAD:
+    case AP_Motors::MOTOR_FRAME_TRI:
+        motorNum = 4;
+        break;
+    case AP_Motors::MOTOR_FRAME_HEXA:
+    case AP_Motors::MOTOR_FRAME_Y6:
+    case AP_Motors::MOTOR_FRAME_SINGLE:
+    case AP_Motors::MOTOR_FRAME_COAX:
+        motorNum = 6;
+        break;
+    case AP_Motors::MOTOR_FRAME_OCTA:
+    case AP_Motors::MOTOR_FRAME_OCTAQUAD:
+        motorNum = 8;
+        break;
+    case AP_Motors::MOTOR_FRAME_DODECAHEXA:
+        motorNum = 12;
+        break;
+
+    case AP_Motors::MOTOR_FRAME_UNDEFINED:
+        break;
+    case AP_Motors::MOTOR_FRAME_TAILSITTER:
+        break;
+    case AP_Motors::MOTOR_FRAME_HELI_DUAL:
+        break;
+    case AP_Motors::MOTOR_FRAME_HELI_QUAD:
+        break;
+    default:
+        break;
+    }
+
+    // motors count
+    int8_t motorCount = 0;
+    for (uint8_t i = 0; i < NUM_SERVO_CHANNELS; i++) {
+        SRV_Channel *chan = SRV_Channels::srv_channel(i);
+        if (chan->is_motor(chan->get_function())) {
```

Comment:
This is the wrong way to tell if a vehicle is configured. You actually need to be checking that a quadcopter has motors 1, 2, 3, 4 mapped (and you don't care if there are extra), but the check as you implemented it would arm with just motors 2, 2, 2, 2 which obviously won't be able to fly correctly.

Reply:
Easiest fix here is probably to have an array+array_len of required servo functions based on the frame type, and implement either a `SRC_Channel::get_first_channel_for_function` (mirroring `RC_Channel::get_channel_for_option`), or (probably better) a `bool SRV_Channel::function_is_assigned`